src/smtp.nim

```diff
@@ -55,12 +55,19 @@ export Port
 
 type
   Message* = object
-    msgTo: seq[string]
-    msgCc: seq[string]
+    msgTo: seq[Email]
+    msgCc: seq[Email]
+    msgBcc: seq[Email]
+    msgReplyTo: seq[Email]
+    msgSender: Email
     msgSubject: string
     msgOtherHeaders: StringTableRef
     msgBody: string
 
+  Email* = object
+    name: Option[string]
```

Comment:
Make this of type `string`. Strings already have a None value and it's the empty string.